app/config.py.example

```diff
@@ -12,6 +12,9 @@ MEDIA_DIRECTORY = 'media'
 PICTURE_EXTENSION = '.png'
```

Comment:
Maybe we should use an existing solution for i18n, like [`gettext`](https://docs.python.org/3/library/gettext.html)?

Reply:
If you prefer gettext, I can switch to that. I didn't use it as it seems to me to be a worse architecture design choice as it keeps the text string inside the actual code, while the method I went with has a clear distinction between code and translations. This method is also supported by translation sites such as Parse (which costs money) and isn't something I made up.

However, as I said, if you prefer gettext, let me know and I'll work on that.

Reply:
Parse actually [recommends `gettext`](https://phrase.com/blog/posts/learn-gettext-tools-internationalization/).

`gettext` is just widely supported and the de-facto standard for i18n.
It has a hole ecosystem around it, including various tools and wide community support.
The `.po`, `.pot` and `.mo` filetypes are supported by almost any translation system.

As for having the strings in the code, it can be an advantage as it helps to find the context of the string easily.

Reply:
ok, then I'll change to gettext. 
Just for the record:  https://phrase.com/blog/posts/fastapi-i18n/

Reply:
I've managed to work but I've resorted to hacking it as the [official documentation ](https://jinja.palletsprojects.com/en/2.11.x/extensions/) just does not work. It says to do:
```
env = jinja2.Environment(extensions=["jinja2.ext.i18n"])
env.install_gettext_translations(translations)
```
but this isn't working. I'm getting: Unresolved attribute reference 'install_gettext_translations' for class 'Environment'.
I looked online but couldn't find any solution, and only found this: https://github.com/tiangolo/fastapi/issues/513.

To get it to work I added:

```
templates = Jinja2Templates(directory=TEMPLATES_PATH)
templates.env.add_extension('jinja2.ext.i18n')
he = gettext.translation('base', localedir=os.path.join(APP_PATH, "locales"), languages=['he'])
he.install()
InternationalizationExtension(templates.env)._install(he, newstyle=True)
```

However, "_install()" is a protected member and shouldn't be accessed like this. Any ideas? I'm all out of them after trying everything I could think of.

Reply:
I had great quality time reading the Jinja2 [actual source code](https://github.com/pallets/jinja/blob/master/src/jinja2/ext.py#L169).

It looks like `templates.env.install_gettext_translations` is the `_install` method you're looking for.